docs/source/changelog.rst

```diff
@@ -8,6 +8,119 @@ Changes listed here will be part of the next release.
 
 --------------------------------------------------------------------------------
 
+0.4.0 (2023.04.17)
+******************
+
+This release adds the commands ``get`` and ``unset``, and updates ``new`` and
+``set``.
+It adds and normalizes many flags (``--keys``, ``--message``, ``--depth``,
+``--yes``) and functionalities ("requesting user responses") across all commands
+to give a smooth and predictable user experience.
+A `demo repository <https://github.com/psyinfra/onyo-demo>`__ shows how an
+inventory created and managed with Onyo looks like.
+The code-base and tests use a lot of new features and are uplifted to reflect
+current standards across the project.
+
+The highlights are:
+
+New Commands
+------------
+- add command ``onyo unset``: remove key/value pairs from assets
+- add command ``onyo get``: query the onyo repository
+
+Command Changes
+---------------
+- overhaul of ``onyo new``:
+    - expect full asset path and name as an argument, instead of reading
+      name fields via TUI
+    - allow creation of multiple assets in one call
+    - verify validity of asset name
+    - do not open new assets with an editor by default
+    - add diff-like output after reading/combining all information
+    - add flags:
+        - ``--edit``: open new asset(s) in editor
+        - ``--keys``: set key/value(s) to new asset(s)
+        - ``--path``: list paths for newly created asset(s)
+        - ``--tsv``: read information from table instead of TUI
+        - ``--yes``: answer yes to all prompts
+- normalize flags ``--path`` and ``--keys`` for commands ``get``, ``new``,
+  ``set``, ``unset``
+- add ``onyo set --rename``: rename assets (update pseudo-keys)
+- add ``--message`` flag to all committing commands
+- add to ``onyo edit`` flags ``--yes`` and ``--quiet``
+- add to ``onyo mkdir`` flags ``--yes`` and ``--quiet``
+- remove flag ``onyo set --recursive`` (set and unset operate recursively by
+  default)
+- key/value pairs are now space-separated (rather than comma-separated)
+- for user responses: normalize the text ("Update assets? (y/n)") and behavior
+  (remove default options, user has to explicitly answer) along commands
+- rename template "standard" -> "empty"
+- retire ``.onyo/temp/``, assets are changed in place and changes reversed when
+  needed
+
+API
+---
+- add property ``Repo.templates``: the templates in ``.onyo/`templates``
+- remove property ``Repo.gitfiles``: unused property
```

Comment:
In this case maybe, but I feel like there should be a reason named for changes of properties. E.g. when we remove `Repo.opdir` just "remove property" feels to weak, one would maybe like to know why it was removed and what it did. And with `Repo.gitfiles` it was not even clear what the name meant - so I think specifically mentioning that it was removed because it is unused feels better for me (more so in a year or two when I scroll through this again).

So I have changed the phrase, but leave a reason there.